File: code/349/349_intersection_of_two_arrays.js

/**
 * javascript方案1
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection1 = function(nums1, nums2) {
    var set1 = new Set(nums1);
    var set2 = new Set(nums2);
    var result = new Set([...set1].filter(x => set2.has(x)));
    return [...result]
};

/**
 * javascript方案2
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection2 = function(nums1, nums2) {
    var set1 = new Set(nums1);
    var set2 = new Set(nums2);
    var result = [];
    if(set1.size < set2.size){
        for(let item of set1){
            if(set2.has(item)){
                result.push(item);
            }
        }
        return result;
    }else{
        for(let item of set2){
            if(set1.has(item)){
                result.push(item);
            }
        }
        return result;
    }
};

File: code/349/349_intersection_of_two_arrays.cs
public class Solution1 {
    public int[] Intersection(int[] nums1, int[] nums2) {
        var set1 = new HashSet<int>(nums1);
        set1.IntersectWith(new HashSet<int>(nums2));
        return set1.ToArray();
    }
}

public class Solution2 {
    public int[] Intersection(int[] nums1, int[] nums2) {
        var arr1 = DelRepeatData(nums1);
        var arr2 = DelRepeatData(nums2);
        var result = new List<int>();
        foreach(int item in arr1){
            if(Array.IndexOf(arr2,item) > -1){
                result.Add(item);
            }
        }
        return result.ToArray();
    }

    public int[] DelRepeatData(int[] arr)
    {
        return arr.GroupBy(p => p).Select(p => p.Key).ToArray();
    }
}